open_lm/model.py

```diff
@@ -142,10 +143,19 @@ def __init__(self, layer_id, args: Params):
         self.head_dim = args.dim // args.n_heads
         self.attention = CustomAttn(layer_id, args)
 
-        # this follows llama / lit llama -- go to multiple of 256
-        hidden_dim = 256 * ((int(2 * 4 * args.dim / 3) + 256 - 1) // 256)
 
-        self.feed_forward = xops.SwiGLU(args.dim, hidden_dim, args.dim, bias=False)
+        if args.ffn_type == "swiglu":
+            # this follows llama / lit llama -- go to multiple of 256
+            hidden_dim = 256 * ((int(2 * 4 * args.dim / 3) + 256 - 1) // 256)
+            self.feed_forward = xops.SwiGLU(args.dim, hidden_dim, args.dim, bias=False)
+        elif args.ffn_type == "gelu":
+            # Follows mosaic mpt7b
+            hidden_dim = args.dim * 4
+            self._ff_w1 = nn.Linear(args.dim, hidden_dim, bias=True)
```

Comment:
do we need the bias here?

Reply:
not sure, https://github.com/mosaicml/llm-foundry/blob/db2233e5c660c2c8b2b9973b4de910add1c93428/llmfoundry/models/layers/ffn.py#L27-L46 has it on by default, not sure if we want it on or not

Reply:
I vote to get rid of bias

Reply:
ok done